### ISSUE TITLE
How make effective change of port using property plugin.httpfileupload.announcedWebPort

### ISSUE THREAD

**Issue by ErwanF76** [igniterealtime/openfire-httpFileUpload-plugin#21]
Hi

Already posted some questions in https://github.com/guusdk/httpfileuploadcomponent/issues/26 regarding issues I have to make effective the change of port when using the plug-in plugin.httpfileupload.

I changed the default value 7443 to 7444 using the property plugin.httpfileupload.announcedWebPort in my Openfire environment.
Unfortunately, this port is not opened (even after restart of Openfire). Consequently, I can't upload file from my Converse.js client.

My question here is quite simple:
How can I make effective the change of port ?

Many thx for your help :)
KR
ErwanF

**Comment by ErwanF76** [comment]
Still me ...
I had a look at openfire code to confirm the suspicion I have regarding the association of HTTP ports with certificates.
My guess is that if I changed the port in the plug-in plugin.httpfileupload configuration, the plug-in doesn't update Jetty connection configuration in order to associate the new port with the certificates used by the default ones.

Is my understanding correct ?
How can I push the association ( new-port ; default-port-certificate ) into Jetty ?

Thx again for your feedback. I would appreciate a lot :+1: 
KR

**Comment by deleolajide** [comment]
![image](https://user-images.githubusercontent.com/110731/88277791-35c08380-ccd9-11ea-9d78-2fbc461c0215.png)

**Comment by ErwanF76** [comment]
Hi !

Thx first for taking consideration of my issue.
However, it seems I wasn't enough clear regarding the issue I'm facing. Therefore I need to give more info.
As I described in the ticket guusdk/httpfileuploadcomponent#26, I want to distinct chat flow from file transfer flow.
Actually, I'm using Converse.js which is a web client and use HTTP-bind SSL port 7443 for chat.
Using the plug-in HTTPFileupload, this one uses also the same port.
That's I don't want.
I thought that configuring another port (7444 using the property plugin.httpfileupload.announcedWebPort), it would allow this.
But it doesn't: Openfire is not aware of this port (except if you change the HTTP Bind Settngs as you mentionned, but then I will loose port 7443 for chat).

I was now thinking to adapt the /var/lib/openfire/plugins/httpfileupload/classes/WEB-INF/web.xml configuration of the HTTPFileupload plug-in by adding something like:
    <Call name="addConnector">
        <Arg>
            <New class="org.eclipse.jetty.server.ssl.SslSelectChannelConnector">
                <Arg>
                    <New class="org.eclipse.jetty.http.ssl.SslContextFactory">
                        <Set name="keyStore">/etc/openfire/security/keystore</Set>
                        <Set name="keyStorePassword">OBF:b91cd1a54781790beaa2baf741fa6789</Set>
                        
                        <Set name="trustStore">/etc/openfire/security/truststore</Set>
                        <Set name="trustStorePassword">OBF:b91cd1a54781790beaa2baf741fa6789</Set>
                    </New>
                </Arg>
                <Set name="port">7444</Set>
                <Set name="maxIdleTime">30000</Set>
            </New>
        </Arg>
    </Call>

But this additional config add no effect :(

I had a look at the code of the plug-in
https://github.com/igniterealtime/openfire-httpFileUpload-plugin/blob/master/src/java/org/igniterealtime/openfire/plugins/httpfileupload/HttpFileUploadPlugin.java
------------------------------------------
context = new WebAppContext( null, pluginDirectory.getPath() + File.separator + "classes", "/httpfileupload" );
------------------------------------------
I would expect that it works ...
Any idea ?

Thx a lot again :)
ErwanF

**Comment by deleolajide** [comment]
You would need to add another webserver instance to Jetty for that. There are only two Jetty webserver instances in openfire. One for admin web pages on ports 9090/9091 and the other for http-bind, websockets and other application webservices on ports 7070/7443.

**Comment by ErwanF76** [comment]
Hi :)

This what I tried yesterday, setting-up Nginx (see detail in ticket https://github.com/guusdk/httpfileuploadcomponent/issues/26).
Unfortunately, the HTTP upload failed because the UUID temporary subfolder wasn't created below the httpfileupload context folder.
I guessed yesterday this missing subfolder was created on the fly by the HTTPFileUpload plug-in (and not by the client which is in my case Converse.js). 
The issue could be connected to the fact the I installed Nginx on another LXC than the one Openfire is running on. This may prevent the creation of the UUID temporary sub-folder. It's my assumption at the moment. Any clue regarding this ?

Still many thx for your consideration
ErwanF

PS: another solution would be to implement and additional Java plug-in that would complete the HTTPFileUpload plug-in in order to create another HTTPs connector (with the new port) and add it to the current HTTP-bind jetty instance.
Basically, something I tried by patching the web.xml file of the HTTPFileUpload plug-in.
